p2p/starknet/stream.go

```diff
@@ -0,0 +1,16 @@
+package starknet
+
+// Stream represents a series of messages that can be accessed by invoking Stream a number of times. With the last element in the stream
```

Comment:
Mind finishing off this comment? I'm genuinely interested in knowing what happens at the end :)

Reply:
Is this the iterator implementation you were looking for btw? It's cool how this avoids type assertions. All others that I've seen use an interface with methods like `Next() any` and `Valid() bool`.

Reply:
> Mind finishing off this comment? I'm genuinely interested in knowing what happens at the end :)

Haha, I must have gotten distracted for some reason :smile: It should continue like, "After stream is entirely consumed of elements, it should return false as its second return value"

> Is this the iterator implementation you were looking for btw? It's cool how this avoids type assertions. All others that I've seen use an interface with methods like Next() any and Valid() bool.

Yeah, I was trying to find a more capable implementation actually. But this will do. Stole it from https://github.com/golang/go/discussions/54245#discussion-4277068, I can not take any credit for it :smile:

Reply:
Oh that's an awesome resource. Thanks!